### DISCUSSION TITLE
Dealing with large datasets in neurokit2

### DISCUSSION THREAD

**Discussion by J4NPM** [neuropsychology/NeuroKit#516] [Q&A]
Hi everyone,

I am having issues when dealing with large ECG datasets in neurokit2. 
ecg_peaks() can handle with int8 format ("uncleaned"), but ecg_delineate() cannot, and seems to accept only float64 (such as returned by ecg_clean()). However, with float64 datasets it quickly runs into memory issues.
I also tried running ecg_delineate() on single  heartbeats (such as the epochs returned by ecg_segment() or manually selecting small segments of data) and that also raised errors. 
How to handle large datasets?
Is there a way to work entirely with int8 datasets?
Is there a way to use ecg_delineate() on a single beat?

Thanks for any help!

**Comment by DominiqueMakowski** [comment]
I opened an issue (https://github.com/neuropsychology/NeuroKit/issues/517) as it seems more like a potential bug that we can address :)
